### PR TITLE
JSON metrics output

### PR DESCRIPTION
**Pull request by egor-dolzhenko**
This pull request will:
- Place QC metrics in a JSON file (with extension `.metrics.json`)
- Add STR genotype to the metrics file

Example:
```json
{
  "DMPK": {
    "AlleleDepths": "28.50/21.93",
    "Genotype": "12/112"
  }
}
```

### CODE REVIEW

File: reviewer/metrics/Metrics.cpp [hunk 1]
```diff
@@ -18,20 +18,45 @@
...
+static map<string, Genotype> getGenotypes(const LocusSpecification& locusSpec, const GraphPaths& hapPaths)
+{
+    map<string, Genotype> genotypes;
+    for (const auto& variantSpec : locusSpec.variantSpecs())
+    {
+        assert(variantSpec.nodes().size() == 1);
+        const auto strNode = variantSpec.nodes().front();
+
+        Genotype genotype;
+        for (int hapIndex = 0; hapIndex != hapPaths.size(); ++hapIndex)
```
Comment (ctsa) [inline reviewer/metrics/Metrics.cpp:48]: This could be simplified to range based for.

**Review by ctsa — APPROVED** [review]
Good idea to move the metrics to a json file.

File: reviewer/metrics/Metrics.cpp (hunk 1 above)
Comment (egor-dolzhenko) [reply reviewer/metrics/Metrics.cpp:48]: Good point!

**Comment by ctsa** [comment]
Do we want to force the metrics users to string parse these outputs? Why shouldn't these per-allele values be a list instead of a string separated by "/"?

**Comment by egor-dolzhenko** [comment]
> Do we want to force the metrics users to string parse these outputs? Why shouldn't these per-allele values be a list instead of a string separated by "/"?

Good question. The pros of this format is consistency with EH output and that some users work with these jsons directly in their text editors:
```json
{
  "DMPK": {
    "AlleleDepths": "28.50/21.93",
    "Genotype": "12/112"
  }
}
```
might be a bit simpler to read than this (especially as the number of metrics grows)
```json
{
   "DMPK":{
      "AlleleDepths":[
         28.50,
         21.93
      ],
      "Genotype":[
         12,
         112
      ]
   }
}
```
As you pointed out, the con is that this format is harder to parse. Perhaps it's a good enough reason to switch to it. What do you think?

**Comment by ctsa** [comment]
In general, one of the big points of VCF->JSON is that parsing becomes trivial, so I don't think the readability is really a consideration, but consistency with EH is really important here, I agree for that reason it probably should be left as is.

**Comment by egor-dolzhenko** [comment]
Sounds good!

**Pull request merged**